src/net.rs

```diff
@@ -31,10 +34,23 @@ pub fn ssl_context<C>(cacert: C,
     if let Some(key) = key {
         try!(ctx.set_private_key_file(key.as_ref(), X509FileType::PEM));
     };
-    Ok(Openssl { context: Arc::new(ctx) })
+    Ok(OpensslClient(ctx))
+}
+
+impl hyper::net::SslClient for OpensslClient {
+    type Stream = SslStream<HttpStream>;
+
+    fn wrap_client(&self, stream: HttpStream, host: &str) -> hyper::error::Result<Self::Stream> {
+        let mut ssl = try!(Ssl::new(&self.0));
+        try!(ssl.set_hostname(host));
+        let host = host.to_owned();
+        ssl.set_verify_callback(SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT, move |p, x| openssl_verify::verify_callback(&host, p, x));
```

Comment:
I was wrong about this - the `FAIL_IF_NO_PEER_CERT` only matters for servers - on clients it's ignored.